app/screens/Account/survey.js

```diff
@@ -1,19 +1,59 @@
+import { format } from "date-fns";
 import React, { Component } from "react";
-import {
-  StyleSheet,
-  View,
-  Button,
-  TouchableOpacity,
-  TouchableHighlight
-} from "react-native";
+import { connect } from "react-redux";
+import { View, ListView, TouchableOpacity } from "react-native";
 import Icon from "react-native-vector-icons/MaterialIcons";
 
-import { Text, Wrapper, PercentComplete } from "../../components";
+import {
+  Text,
+  Wrapper,
+  PercentComplete,
+  AnnotationObservation,
+  Map
+} from "../../components";
 import { baseStyles } from "../../styles";
 
+import {
+  pickSurveyType,
+  calculateCompleteness
+} from "../../lib/calculate-completeness";
+
+import { osm, setActiveObservation } from "../../actions";
+
+import { selectActiveSurveys } from "../../selectors";
+
 class SurveysScreen extends Component {
+  componentWillMount() {
+    const { surveys, match: { params: { surveyId } } } = this.props;
+    let { definition: { featureTypes } } = surveys.find(survey => {
+      return survey && survey.definition && survey.definition.name === surveyId;
+    });
+
+    const ds = new ListView.DataSource({
+      rowHasChanged: (r1, r2) => r1 !== r2
+    });
+
+    this.setState({ observations: ds.cloneWithRows([]) });
+
+    osm.getObservationsBySurveyId(surveyId, (err, observations) => {
+      let resultsWithCompleteness = observations.map(observation => {
+        // For each observation, pick the correct fields for that observation's
+        // type and calculate completeness
+        let surveyFields = pickSurveyType(featureTypes, observation);
+        let percentage = calculateCompleteness(surveyFields, observation);
+        return Object.assign({}, observation, { percentage });
+      });
+
+      this.setState({
+        observations: ds.cloneWithRows(resultsWithCompleteness),
+        numObservations: resultsWithCompleteness.length
+      });
+    });
+  }
+
   render() {
-    const { history } = this.props;
+    const { history, match: { params: { surveyId } } } = this.props;
```

Comment:
It looks like we need to get `setActiveObservation` from the props here for it to work in the `onPress` handler on line 106.